### PR TITLE
Feature/yaml validation

### PR DESCRIPTION
**Pull request by live1206**
Return error list instead of throwing exception during Yaml deserialization

### CODE REVIEW

File: src/docfx/lib/YamlUtility.cs [hunk 1]
```diff
@@ -100,7 +100,7 @@ public static (List<Error> erros, JToken token) Deserialize(TextReader reader)
             }
             catch (YamlException ex)
             {
-                errors.Add(Errors.YamlSyntaxError(ex));
+                errors.Add(Errors.YamlSyntaxError(ex.InnerException));
```
Comment (yufeih) [inline src/docfx/lib/YamlUtility.cs]: you want to test other exceptions, they may not have an inner exception

File: src/docfx/lib/markdown/ExtractYamlHeader.cs [hunk 2]
```diff
@@ -24,22 +25,20 @@ internal static class ExtractYamlHeader
...
+                        {
+                            result.Value = obj;
                         }
-                        catch (Exception ex)
+                        else
+                        {
+                            errors.Add(Errors.YamlHeaderNotObject(file, isArray: yamlHeaderObj is JArray));
+                        }
+
+                        if (yamlErrors.Any())
```
[the remark this answers is not in the document]
Comment (live1206) [reply src/docfx/lib/markdown/ExtractYamlHeader.cs]: right

File: test/docfx.Test/lib/YamlUtilityTest.cs [hunk 3]
```diff
@@ -277,8 +289,11 @@ public void TestDuplicatedKeys()
 Key1: 0
 Key1: 0
 ";
-            var exception = Assert.Throws<YamlException>(() => YamlUtility.Deserialize(yaml));
-            Assert.Equal("An item with the same key has already been added. Key: Key1", exception.InnerException.Message);
+            var (errors, _) = YamlUtility.Deserialize(yaml);
+            Assert.NotEmpty(errors);
```
[the remark this answers is not in the document]
Comment (live1206) [reply test/docfx.Test/lib/YamlUtilityTest.cs]: nice one

File: src/docfx/lib/YamlUtility.cs (hunk 1 above)
Comment (live1206) [reply src/docfx/lib/YamlUtility.cs]: true, still use exception and assert contain in the test

**Pull request merged**